File: api/src/Domain/ProductCatalog/ValueObjects/PadSize.cs
using System;
using DetailingArsenal.Domain;

namespace DetailingArsenal.Domain.ProductCatalog {
    public class PadSize : Entity<PadSize> {
        public Measurement Diameter { get; set; }
        public Measurement? Thickness { get; set; }

        public PadSize(Measurement diameter, Measurement? thickness = null) {
            Id = Guid.NewGuid();
            Diameter = diameter;
            Thickness = thickness;
        }

        public PadSize(Guid id, Measurement diameter, Measurement? thickness = null) {
            Id = id;
            Diameter = diameter;
            Thickness = thickness;
        }
    }
}

File: app/src/api/admin/data-transfer-objects/pad-size.ts
import { Measurement } from '@/api/shared/data-transfer-objects/measurement';
import { MeasurementUnit } from '../../shared/data-transfer-objects/measurement-unit';

export class PadSize {
    constructor(public id: string, public diameter: Measurement, public thickness?: Measurement) {}

    static isThin(p: PadSize) {
        if (p.thickness == null) {
            return false;
        }

        return (
            (p.thickness.amount < 0.75 && p.thickness.unit == MeasurementUnit.Inches) ||
            (p.thickness?.amount < 19.05 && p.thickness?.unit == MeasurementUnit.Millimeters)
        );
    }
}
